sql/core/src/main/scala/org/apache/spark/sql/execution/WholeStageCodegenExec.scala

```diff
@@ -372,7 +374,7 @@ case class WholeStageCodegenExec(child: SparkPlan) extends UnaryExecNode with Co
     try {
       CodeGenerator.compile(cleanedSource)
     } catch {
-      case e: Exception if !Utils.isTesting && sqlContext.conf.wholeStageFallback =>
+      case e: JaninoRuntimeException if !Utils.isTesting && sqlContext.conf.wholeStageFallback =>
```

Comment:
This change is useless. `CodeGenerator.doCompile` catches `Exception` and re-throw `Exception`.